controllers/tenant/namespaces.go

```diff
@@ -73,6 +73,25 @@ func (r *Manager) syncNamespaceMetadata(ctx context.Context, namespace string, t
 				}
 			}
 
+			if tnt.Spec.NamespaceOptions != nil && len(tnt.Spec.NamespaceOptions.AdditionalMetadataList) > 0 {
+				for _, additionalMetadata := range tnt.Spec.NamespaceOptions.AdditionalMetadataList {
+					var namespaceSelectedBySelector bool
+
+					namespaceSelectedBySelector, err = utils.IsNamespaceSelectedBySelector(ns, additionalMetadata.NamespaceSelector)
+					if err != nil {
+						return err
+					}
+
+					if !namespaceSelectedBySelector {
+						continue
+					}
+
+					for k, v := range additionalMetadata.Labels {
+						labels[k] = v
+					}
+				}
+			}
+
```

Comment:
Initialize the labels empty on L59

```
labels:= make(map[string]string)
```

Add default labels here, to ensure they were not overwriten by spec:
```suggestion
			labels["kubernetes.io/metadata.name"] = namespace
			labels[capsuleLabel] =  tnt.GetName()
```

Reply:
Made the change, also added a test to check this